Gifski/MainWindowController.swift

```diff
@@ -301,6 +306,18 @@ final class MainWindowController: NSWindowController {
 		}
 	}
 
+	@IBAction private func quickLook(_ sender: Any) {
+		guard let panel = QLPreviewPanel.shared() else {
+			return
+		}
+
+		if panel.isVisible {
+			panel.orderOut(nil)
+		} else {
+			panel.makeKeyAndOrderFront(nil)
+		}
```

Comment:
The above could be a `QLPreviewPanel#toggle()` extension so the code in here could simply be:

```swift
QLPreviewPanel.shared()?.toggle()
```

Reply:
Oh nice didn't think of that. Makes sense here. Though, as I learned painfully by experience, we should be prefixing extension methods on Apple-provided classes to avoid unexpected behaviors. Do you have a preference for a prefix, like `gifski_toggle()`? Or do you have a different method for that type of additions?

Reply:
I don't see the need for a prefix. If Apple introduces a `.toggle()` method, although very unlikely, it will need change the semantics of an already compiled binary. It might cause a ambiguity / conflict error at compile-time, but that's ok.

Reply:
This is not the only case I was thinking of. I’ve added a property once, `aspectRatio` on PHAsset I believe, and even though there was no method with that name, we’ve got undefined behavior in runtime. App worked most of the time correctly, but there was a case where our method overshadowed an existing, private one and it was doing lots of different, unexpected things with the image.

Reply:
Anyhow, I've added an extension to `QLPreviewPanel` without any prefix for now.

Reply:
I've heard of the problem with Objective-C, but never with Swift.

Reply:
It happened in Swift as well! Makes sense as you can have dynamic/message dispatch in Swift as well.